src/cipher.jl

```diff
@@ -71,6 +73,25 @@ function get_cipher_types()
     return _cipher_types
 end
 
+const _cipher_suites = [:CBC] # [:CBC, :GCM, :CCM]
+
+function gen_key32_iv16(pw::Array{UInt8,1}, salt::Array{UInt8,1})
+    s1 = digest("MD5", [pw; salt])
+    s2 = digest("MD5", [s1; pw; salt])
+    s3 = digest("MD5", [s2; pw; salt])
+    return ([s1; s2], s3)
+end
+
+function add_padding_PKCS5(data::Array{UInt8,1}, block_size::Int)
+  padlen = block_size - (sizeof(data) % block_size)
+  # return [data; map(i -> UInt8(padlen), 1:padlen)]
+  return [data; convert(Array{UInt8,1}, map(i -> padlen, 1:padlen))] # to pass test julia 0.3
+end
+
+function trim_padding_PKCS5(data::Array{UInt8,1})
+  padlen = data[sizeof(data)]
+  return data[1:sizeof(data)-padlen]
+end
```

Comment:
I think that padding argolithms will be selectable and should be independent of cipher algorithms, so we should not make hard binding automatically.